mods/ra/maps/soviet-07/soviet07.lua

```diff
@@ -0,0 +1,298 @@
+		cameraCCTrigger = true
+	end
+	Utils.Do(CCGuards, function(actor)
+		actor.Hunt()
+	end)
+end)
+
+Trigger.OnKilled(PBoxBrl, function()
+	PBox.Kill()
+	Utils.Do(Dogs, function(actor)
+		actor.Owner = player
+	end)
+	player.MarkCompletedObjective(sovietObjective6)
+end)
+
+Trigger.OnKilled(PrisonEntranceGuard, function()
+	if controlCenterTrigger then
+		Utils.Do(PrisonerGuards, function(actor)
+			actor.Hunt()
+		end)
+	end
+end)
+
+IntroSequence = function()
+	StartingUnits = Reinforcements.Reinforce(player, StartingUnitsReinforcements, { StartingUnitsSpawn.Location, SoldierTrap1Waypoint1.Location }, 0)
+	Trigger.AfterDelay(DateTime.Seconds(3), function()
+		Utils.Do(EntranceGuards, function(actor)
+			if not SoldierTrap1.IsDead then
+				actor.Attack(SoldierTrap1)
+			end
+			actor.AttackMove(SoldierTrap1Waypoint1.Location)
+			actor.AttackMove(SoldierTrap1Waypoint2.Location)
+			actor.AttackMove(SoldierTrap1Waypoint3.Location)
+		end)
+		Media.PlaySpeechNotification(player, "TimerStarted")
+		timerStarted = true
+	end)
+	Trigger.OnAllKilled(StartingUnits, function()
+		enemy.MarkCompletedObjective(alliedObjective)
+	end)
+end
+
+WorldLoaded = function()
+	player = Player.GetPlayer("USSR")
+	enemy = Player.GetPlayer("Greece")
+	Camera.Position = SoldierTrap1Waypoint1.CenterPosition
+	Actor.Create("camera", true, { Owner = player, Location = CameraStart1.Location })
+	Actor.Create("camera", true, { Owner = player, Location = CameraStart2.Location })
+	IntroSequence()
+	Trigger.OnObjectiveAdded(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "New " .. string.lower(p.GetObjectiveType(id)) .. " objective")
+	end)
+	Trigger.OnObjectiveCompleted(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective completed")
+	end)
+	Trigger.OnObjectiveFailed(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective failed")
+	end)
+	Trigger.OnPlayerWon(player, function()
+		Media.PlaySpeechNotification(player, "Win")
+	end)
+	Trigger.OnPlayerLost(player, function()
+		Media.PlaySpeechNotification(player, "Lose")
+	end)
+	alliedObjective = enemy.AddPrimaryObjective("Destroy all Soviet troops.")
+	sovietObjective1 = player.AddPrimaryObjective("Deactivate the security system.")
+	sovietObjective2 = player.AddPrimaryObjective("Rescue the engineers.")
+	sovietObjective3 = player.AddPrimaryObjective("Get the engineers to the coolant stations.")
+	sovietObjective4 = player.AddPrimaryObjective("Use an Engineer to reprogram the security system.")
+	sovietObjective5 = player.AddPrimaryObjective("Get an Engineer to the reactor core.")
+	sovietObjective6 = player.AddSecondaryObjective("Free the dogs.")
+end
+
+Tick = function()
+	if player.HasNoRequiredUnits() and timerStarted then
+		enemy.MarkCompletedObjective(alliedObjective)
+	end
+	if remainingTime/25 == 300 and not Map.Difficulty == "Hard" then
```

Comment:
What about `remainingTime == DateTime.Seconds(300)` instead of `remainingTime/25 == 300`?
(Also affects below.)